ldap-api/src/main/java/ca/bc/gov/hlth/ldapapi/service/LdapService.java

```diff
@@ -49,25 +47,27 @@ public User authenticate(UserCredentials userCredentials) {
         }
 
         boolean userUnlocked = checkUserLocked(userInfo.getAttributes());
-        boolean userPasswordExpired = checkUserPasswordExpired(userInfo.getAttributes());
 
         boolean validCredentials = false;
-        if (userUnlocked) {
-            validCredentials = authenticateUser(userInfo.getName(), userCredentials.getPassword());
-        }
-
-        // Get nbLoginAttempts and lockoutTimeOut after authenticateUser 
         long lockoutTimeInHours = 0;
         int remainingAttempts = 3;
-        if (!validCredentials){
-            LoginAttempts loginAttemptsForUser = loginAttemptsMap.get(userInfo.getName());
-            if (loginAttemptsForUser != null){
-                remainingAttempts = 3 - loginAttemptsForUser.getAttempts();
-             
-                lockoutTimeInHours = ChronoUnit.HOURS.between(loginAttemptsForUser.getLastAttempt(), LocalDateTime.now());
+
+        if (userUnlocked) {
+            AuthenticationResult authenticationResult = authenticateUser(userInfo.getName(), userCredentials.getPassword());
+            validCredentials = authenticationResult.isAuthenticated;
+            userUnlocked = !authenticationResult.isLocked;
+
+            // Get nbLoginAttempts and lockoutTimeOut after authenticateUser
+            if (!validCredentials) {
+                LoginAttempts loginAttemptsForUser = loginAttemptsMap.get(userInfo.getName());
+                if (loginAttemptsForUser != null) {
+                    remainingAttempts = 3 - loginAttemptsForUser.getAttempts();
+                    lockoutTimeInHours = ChronoUnit.HOURS.between(loginAttemptsForUser.getLastAttempt(), LocalDateTime.now());
```

Comment:
I moved this code inside the `if (userUnlocked)` because there's no point in tracking attempts and lockout time if the account is locked and we're already skipping authentication.